courses/sync_external_courses/utils.py

```diff
@@ -0,0 +1,202 @@
+
+    def __init__(self):
+        self.stats = {
+            "courses_created": StatItemsCollection(
+                "courses_created", "External Course Codes"
+            ),
+            "existing_courses": StatItemsCollection(
+                "existing_courses",
+                "External Course Codes",
+            ),
+            "course_runs_created": StatItemsCollection(
+                "course_runs_created",
+                "External Course Run Codes",
+            ),
+            "course_runs_updated": StatItemsCollection(
+                "course_runs_updated",
+                "External Course Run Codes",
+            ),
+            "course_runs_without_prices": StatItemsCollection(
+                "course_runs_without_prices",
+                "External Course Run Codes",
+            ),
+            "course_runs_skipped": StatItemsCollection(
+                "course_runs_skipped",
+                "External Course Run Codes",
+                display_name="Course Runs Skipped due to bad data",
+            ),
+            "course_runs_expired": StatItemsCollection(
+                "course_runs_expired",
+                "External Course Run Codes",
+                display_name="Expired Course Runs",
+            ),
+            "course_runs_deactivated": StatItemsCollection(
+                "course_runs_deactivated",
+                "External Course Run Codes",
+            ),
+            "course_pages_created": StatItemsCollection(
+                "course_pages_created",
+                "External Course Codes",
+            ),
+            "course_pages_updated": StatItemsCollection(
+                "course_pages_updated",
+                "External Course Codes",
+            ),
+            "products_created": StatItemsCollection(
+                "products_created",
+                "Course Run courseware_ids",
+            ),
+            "product_versions_created": StatItemsCollection(
+                "product_versions_created",
+                "Course Run courseware_ids",
+            ),
+            "certificates_created": StatItemsCollection(
+                "certificates_created",
+                "Course Readable IDs",
+                display_name="Certificate Pages Created",
+            ),
+            "certificates_updated": StatItemsCollection(
+                "certificates_updated",
+                "Course Readable IDs",
+                display_name="Certificate Pages Updated",
+            ),
+        }
+
+    def add_stat(self, key, code, title=None, msg=None):
+        """
+        Add an item to a specific stat
+        """
+        if key in self.stats:
+            self.stats[key].add(code, title, msg)
+
+    def add_bulk(self, key, codes):
+        """
+        Add multiple items within the same stat
+        """
+        if key in self.stats:
+            for code in codes:
+                self.add_stat(key, code)
```

Comment:
Do we really need these key checks? I think we don't because, in future if someone makes a change and does some kind of typo in key name, this will bypass that instead of an error.